Add support for enum values in NatSpec (#15956)
ethereum/solidity: Changelog.md

Reviewer: The tags allowed for enum values should be documented in [Natspec Format > Tags](https://docs.soliditylang.org/en/develop/natspec-format.html#tags).

Unless it is assumed to be covered by `enum`? I guess struct fields allow Natspec as well and they are not explicitly listed there. On the other hand we don't capture Natspec in AST for struct fields but we now do for enum values so the situation is not the same.

Reviewer: Enums are also not mentioned at all in this bit:

> Documentation is inserted above each `contract`, `interface`, `library`, `function`, and `event` using the Doxygen notation format

Compiler Features:

> * NatSpec: Add support for NatSpec documentation in ``enum`` value definitions.

Reviewer: I see that this is how we were describing such additions in the past, but "support" is quite ambiguous. This Natspec will not show up in `--userdoc` or `--devdoc` output. Using the tags also wasn't even disallowed on enum values before. The only thing that changes is that it's now captured in the AST. How about we say just that?

```suggestion
* NatSpec: Capture Natspec documentation of `enum` values in the AST.
```

Perhaps this should also be stated in the docs. Right now they're rather vague and even incorrect, depending on how you define "support". After reading them and checking this PR I'm really confused about what we actually support.
